### PR TITLE
Convert hashicorp/waypoint-plugin-sdk to GitHub Actions

### PR DESCRIPTION
**Pull request by hc-github-team-es-release-engineering**
Pipeline migrated from [CircleCI](https://app.circleci.com/pipelines/github/hashicorp/waypoint-plugin-sdk) :tada:

## Manual steps

Perform the following steps to complete the migration:

### hashicorp/waypoint-plugin-sdk/go-tests
- [x] Ensure environment variable is updated: `SLACK_WEBHOOK`

- the above `SLACK_WEBHOOK` is likely not needed and will be replaced with a GHA

### CODE REVIEW

File: .github/actions/notify_main_failure/action.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: notify_main_failure
```
Comment (sarahethompson) [inline .github/actions/notify_main_failure/action.yml:1]: I don't think this is being used at all in the new GHA workflows? If you want to still send slack notifications, these are the ways you can do this in GitHub Actions - https://github.com/hashicorp/GHA-migration/issues/41

File: .github/workflows/go-tests.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        # version, because "goenv" can react to it automatically.
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "go-version=$(cat .go-version)" >> "$GITHUB_OUTPUT"
+      - name: Set up Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: "${{ steps.get-go-version.outputs.go-version }}"
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
```
Comment (sarahethompson) [inline .github/workflows/go-tests.yml:84]: you'll need to sha pin this to the latest version

Comment (catsby) [reply .github/workflows/go-tests.yml]: Done in c1b77ec thanks!

File: .github/actions/notify_main_failure/action.yml (hunk 1 above)
Comment (sarahethompson) [reply .github/actions/notify_main_failure/action.yml:1]: @dekimsey just pointed me to this action (https://github.com/hashicorp/actions-slack-status) which should work as a direct replacement for this

Comment (catsby) [reply .github/actions/notify_main_failure/action.yml]: Thanks! We're removing this in favor of your suggestion

**Comment by dekimsey** [comment]
I defined SLACK_WEBHOOK_URL and pointed it to `#feed-waypoint-gh`

**Review by dekimsey — APPROVED** [review]
LGTM

File: .github/workflows/go-tests.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            --disable-all \
+            --timeout 10m \
+            --enable gofmt \
+            --enable gosimple \
+            --enable govet \
+            --verbose
+          skip-pkg-cache: true
+          skip-build-cache: true
+      - uses: hashicorp/actions-slack-status@v1
+        if: failure() && github.ref == 'refs/heads/main'
```
Comment (emilymianeil) [inline .github/workflows/go-tests.yml:28]: Do we only want to report failures on `main` and not on other branches (ie in pull requests)?

**Review by emilymianeil — APPROVED** [review]

**Pull request merged**